sw/device/silicon_creator/manuf/lib/sram_exec_test.c

```diff
@@ -0,0 +1,82 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/status.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_uart_t uart0;
+static dif_pinmux_t pinmux;
+static dif_otp_ctrl_t otp;
+
+static const uint8_t kNumDeviceId = 8;
+
+/**
+ * Read and return 32-bit OTP data via the DAI interface.
+ */
+static status_t otp_ctrl_dai_read_32(const dif_otp_ctrl_t *otp,
+                                     dif_otp_ctrl_partition_t partition,
+                                     uint32_t address, uint32_t *buf) {
+  TRY(dif_otp_ctrl_dai_read_start(otp, partition, address));
+  TRY(otp_ctrl_testutils_wait_for_dai(otp));
+  TRY(dif_otp_ctrl_dai_read32_end(otp, buf));
+  return OK_STATUS();
+}
+
+status_t sram_main() {
+  if (kDeviceType != kDeviceSimDV) {
+    // Configure the pinmux.
+    TRY(dif_pinmux_init(
+        mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+    pinmux_testutils_init(&pinmux);
+
+    // Initialize UART.
+    TRY(dif_uart_init(
+        mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart0));
+    TRY(
+        dif_uart_configure(&uart0, (dif_uart_config_t){
+                                       .baudrate = kUartBaudrate,
+                                       .clk_freq_hz = kClockFreqPeripheralHz,
+                                       .parity_enable = kDifToggleDisabled,
+                                       .parity = kDifUartParityEven,
+                                       .tx_enable = kDifToggleEnabled,
+                                       .rx_enable = kDifToggleEnabled,
+                                   }));
+    base_uart_stdout(&uart0);
```

Comment:
could you use the `ottf_console` [lib](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_framework/ottf_console.c) here?